src/main/kotlin/org/jetbrains/teamcity/maven/sdk/TeamCitySDK.kt

```diff
@@ -81,10 +83,36 @@ public class ReloadJSPMojo() : AbstractTeamCityMojo() {
     }
 }
 
-@Mojo(name = "reload", aggregator = true)
+@Mojo(name = "reloadAgent", aggregator = true)
 public class ReloadPluginMojo() : AbstractTeamCityMojo() {
     override fun doExecute() {
         uploadPluginAgentZip()
         log.info("Plugin uploaded. Wait for TeamCity agent to upgrade.")
     }
 }
+
+@Mojo(name = "reload", aggregator = true)
+class ReloadPluginInRuntimeMojo : AbstractTeamCityMojo() {
+    override fun doExecute() {
+        var doOfflineReload = false;
+        val split = teamcityVersion.split(".")
+        if (split.size < 2 || split[0].toInt() < 2018 || split[1].toInt() < 2) {
+            log.info("Cannot reload plugin in runtime for TeamCity version less then 2018.2. Will perform reloadAgent. ")
```

Comment:
A user potentially do not know, what a `reloadAgent` is. Maybe we could just tell him we are updating Agent-side plugin part? Same applies to next message (line 103)

Reply:
Rewrote the message to make it clearer